File: 2º/PDOO/ProyectoRUBY/lib/s_por_jugador.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative "jugador.rb"
require_relative"sorpresa.rb"

module Civitas
  class S_porJugador<Sorpresa
    def initialize(valor, texto)
      super(texto)
      @valor=valor
    end
    def aplicarAJugador(actual, todos)
      super.informe(actual, todos)
      te="Pagar al jugador "+actual
      s=Sorpresa.new(TipoSorpresa.PAGARCOBRAR, (@valor*(-1)), te);
      for i in todos.size
        if(i==actual)then
        else
            s.aplicarAJugador_pagarCobrar(i, todos);
        end
      end
      s.setValor(@valor*(todos.size()-1));
      for i in todos.size
        if(i==actual)then
          s.aplicarAJugador_pagarCobrar(i, todos);
        else
        end
      end
    end
    def to_s
      aux=super
      return aux
    end  
  end
end

File: 2º/PDOO/ProyectoJAVA/src/civitas/S_porJugador.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author nikitastetskiy
 */
public class S_porJugador extends Sorpresa{
    
    private int valor;
    
    S_porJugador(int valor, String texto){
        super(texto);
        this.valor = valor;
    }
    
    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)) {
            informe(actual, todos);
            Sorpresa s1 = new S_pagarCobrar(valor * -1, "PagarTodos");
            Sorpresa s2 = new S_pagarCobrar(valor * (todos.size() - 1), "PagarActual");
            for (int i = 0; i < todos.size(); i++) {
                if (i != actual) {
                    s1.aplicarAJugador(i, todos);
                }
            }
            s2.aplicarAJugador(actual, todos);
        }
    }
    
    @Override
    public String toString() {
       return super.toString();
    }
}
